pkg/config/piped.go

```diff
@@ -969,14 +974,23 @@ func (n *NotificationReceiverSlack) Validate() error {
 	if len(mentionedAccounts) > 0 {
 		n.MentionedAccounts = mentionedAccounts
 	}
-	if n.HookURL != "" && (n.OAuthToken != "" || n.ChannelID != "") {
-		return errors.New("only one of hookURL or oauthToken and channelID should be used")
+	if n.HookURL != "" && (n.OAuthToken != "" || n.OAuthTokenFile != "" || n.OAuthTokenData != "" || n.ChannelID != "") {
+		return errors.New("only one of hookURL or OAuthToken, OAuthTokenData, OAuthTokenFile and channelID should be used")
 	}
 	if n.HookURL != "" {
 		return nil
 	}
-	if n.OAuthToken == "" || n.ChannelID == "" {
-		return errors.New("missing oauthToken or channelID configuration")
+	if (n.HookURL == "" && n.ChannelID == "") || (n.OAuthToken == "" && n.OAuthTokenFile == "" && n.OAuthTokenData == "") {
+		return errors.New("either HookURL or OAuthToken and OAuthTokenData, OAuthTokenFile and channelID must be set")
```

Comment:
```suggestion
		return errors.New("missing channelID or OAuth token configuration")
```
nits, for less confusing